modules/cats-effect/rules/src/main/scala/org/typelevel/fix/IORandomUUID.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.fix
+
+import scala.meta.*
+import scalafix.v1.*
+
+class IORandomUUID extends SemanticRule("TypelevelIORandomUUID") {
+  def checkSignature(parent: Stat, tree: Term)(implicit doc: SemanticDocument): Patch = {
+    tree.symbol.info.flatMap { info =>
+      info.signature match {
+        case MethodSignature(_, _, TypeRef(_, sym, _))
+            if IOSym.matches(sym) && parent.toString().equals(ParentRandomWithBlocking) =>
+          Some(Patch.replaceTree(parent, "IO.randomUUID"))
+        case _ =>
+          None
+      }
+    }.asPatch
+  }
+
+  def checkDiscardedStat(outer: Stat)(implicit doc: SemanticDocument): Patch = {
+    def checkInner(stat: Stat): Patch = {
+      lazy val self: PartialFunction[Stat, Patch] = {
+        case Term.Apply.Initial(_ @Term.Name(_), _) if outer.toString().equals(OuterMatch) =>
+          Patch.replaceTree(outer, "IO.randomUUID").atomic
+        case Term.Select(_, prop @ Term.Name(_)) =>
+          checkSignature(outer, prop)
+        case Term.Apply.Initial(Term.Select(_, method), _) =>
+          checkSignature(outer, method)
+      }
+
+      self.lift(stat).asPatch
+    }
+
+    checkInner(outer)
+  }
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      case Defn.Val(_, _, _, expr) if expr.toString().equals(OuterMatch) =>
+        Patch.replaceTree(expr, "IO.randomUUID").atomic
+      case Template.Initial(_, _, _, stats) =>
+        stats.map(checkDiscardedStat).asPatch
+      case Term.ForYield(enums, _) =>
+        enums.collect { case Enumerator.Generator(_, rhs) =>
+          checkDiscardedStat(rhs)
+        }.asPatch
+    }.asPatch
+  }
+
+  val IOSym                    = SymbolMatcher.exact("cats/effect/IO#")
+  val ParentRandomWithBlocking = "IO.blocking(UUID.randomUUID())"
```

Comment:
Does this have false positives if there is a `UUID` from a different package?  It's unlikely, and would also have to have a `randomUUID()` method, so it's a more theoretical question.  I just thought semantic rules might use something that looks more like line 65 all the way down.

This is more a question from my ignorance than a practical concern.

Reply:
That's actually a great question! So I'm not completely sure, this is kinda a new territory for me. I suspect there might be a more robust symbol match that could be added (that's still a little black magic to me). I'm gonna give it another look because I'm curious.

Reply:
@rossabaker I actually slept on it and took a crack this morning, I updated my pr with 1) more tests 2) exclusively utilizing `SymbolMatcher.exact`